pkg/kapis/notification/v2/register.go

```diff
@@ -0,0 +1,188 @@
+/*
+
+  Copyright 2020 The KubeSphere Authors.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+package v2
+
+import (
+	"github.com/emicklei/go-restful"
+	openapi "github.com/emicklei/go-restful-openapi"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/kubernetes"
+	"kubesphere.io/kubesphere/pkg/api"
+	"kubesphere.io/kubesphere/pkg/apiserver/query"
+	"kubesphere.io/kubesphere/pkg/apiserver/runtime"
+	kubesphere "kubesphere.io/kubesphere/pkg/client/clientset/versioned"
+	"kubesphere.io/kubesphere/pkg/constants"
+	"kubesphere.io/kubesphere/pkg/informers"
+	"kubesphere.io/kubesphere/pkg/server/errors"
+	"net/http"
+)
+
+const (
+	GroupName      = "notification.kubesphere.io"
+	KeyOpenAPITags = openapi.KeyOpenAPITags
+)
+
+var GroupVersion = schema.GroupVersion{Group: GroupName, Version: "v2"}
+
+func AddToContainer(
+	container *restful.Container,
+	informers informers.InformerFactory,
+	k8sClient kubernetes.Interface,
+	ksClient kubesphere.Interface) error {
+
+	ws := runtime.NewWebService(GroupVersion)
+	h := newNotificationHandler(informers, k8sClient, ksClient)
+
+	// apis for secrets
```

Comment:
KubeSphere doesn't have APIs for secrets already and we still need to add it for notification?

Reply:
The secrets of config and receiver are in the `KubeSphereNotificationNamespace`, and the regular users do not have access to `KubeSphereNotificationNamespace`, so a separate API needs to be provided.